gensim/models/indexer.py

```diff
@@ -0,0 +1,246 @@
+
+        Args:
+            file_name:   Output file path
+
+        Returns:
+            Returns true on success. Otherwise, returns false
+        """
+        pass
+
+    def load(self, file_name):
+        """
+        Load previously dumped info
+
+        Args:
+            file_name:  Input file paht
+
+        Returns:
+            Returns True on success. Otherwise, returns False
+        """
+        pass
+
+
+class AnnoyIndexer(Indexer):
+    """
+    An Annoy-based indexer.
+    For more information about Annoy, please visit https://github.com/spotify/annoy
+    """
+
+    def __init__(self, model=None, **kwargs):
+        """
+        Constructor
+
+        Args:
+            features_num: Number of features
+            tree_size: Number of trees. More trees gives higher precision when querying.
+
+        Raises:
+            TypeError: not provide enough parameters 
+        """
+        if "features_num" not in kwargs:
+            raise TypeError("Must provide parameter features_num")
+        self.features_num = kwargs["features_num"]
+
+        if "tree_size" not in kwargs:
+            raise TypeError("Must provide parameter tree_size")
+        self.tree_size = kwargs["tree_size"]
+
+        self.annoy_indexer = AnnoyIndex(self.features_num)  # Length of item vector that will be indexed
+        self.index_label_dict = {}
+        self.label_index_dict = {}
+        self.id = 0
+
+        # call parent constructor
+        super(AnnoyIndexer, self).__init__(model, **kwargs)
+
+    def add_item(self, label, vector):
+        """
+        Add an item to index
+
+        Args:
+            label: the label of the item (must be unique).
+            vector: the item's vector.
+        """
+        self.index_label_dict[self.id] = label
+        self.label_index_dict[label] = self.id
+        self.annoy_indexer.add_item(self.id, vector)
+        self.id += 1
+
+    def get_item(self, label):
+        """
+        Get an item's vector by its label
+
+        Args:
+            label: The label
+
+        Returns:
+            The item's vector if have. Otherwise, returns None
+        """
+        if label not in self.label_index_dict:
+            return None
+        return self.annoy_indexer.get_item_vector(self.label_index_dict[label])
+
+    def start_indexing(self):
+        """
+        Start the indexing operation.
+        This method is supposed to run after all items are added
+        It may take time to finish.
+        """
+        self.annoy_indexer.build(self.tree_size)
+
+    def get_nearest_items(self, vector, top_n=10):
+        """
+        Get nearest items for an item described by its vector
+
+        Args:
+            vector: The vector
+            top_n: Number of nearest items to get
+
+        Returns:
+            A list of tuples (label, similarity) for the nearest ones
+        """
+        nearest = self.annoy_indexer.get_nns_by_vector(vector, top_n, include_distances=True)
```

Comment:
Can you unpack the `nearest` tuple into named vars? `nearest[0]` and `nearest[1]` are too opaque.